src/reducers/trace.js

```diff
@@ -18,35 +18,52 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-import Immutable from 'immutable';
+import keyBy from 'lodash/keyBy';
 import { handleActions } from 'redux-actions';
 
-import * as jaegerApiActions from '../actions/jaeger-api';
+import { fetchTrace, searchTraces } from '../actions/jaeger-api';
 import { enforceUniqueSpanIds } from '../selectors/trace';
 
-export const initialState = Immutable.Map({
-  traces: Immutable.Map(),
+const initialState = {
+  traces: {},
   loading: false,
   error: null,
-});
+};
+
+function fetchStarted(state) {
+  return { ...state, loading: true };
+}
+
+function fetchTraceDone(state, { meta, payload }) {
+  const trace = enforceUniqueSpanIds(payload.data[0]);
+  const traces = { ...state.traces, [meta.id]: trace };
+  return { ...state, traces, loading: false };
+}
+
+function fetchTraceErred(state, { meta, payload }) {
+  const traces = Object.assign({}, state.traces, { [meta.id]: payload });
```

Comment:
I find it easier to read. For comarison, [the diff](https://github.com/uber/jaeger-ui/pull/48/commits/661359d086e2b97ce2cb48d6272c44051eeb533b).

Additional discussion on it from the redux crew: [docs](http://redux.js.org/docs/recipes/UsingObjectSpreadOperator.html)